examples/her/train_dqn_bit_flip.py

```diff
@@ -0,0 +1,252 @@
+import argparse
+
+import gym
+import gym.spaces as spaces
+import numpy as np
+import torch
+import torch.nn as nn
+
+from pfrl import agents, experiments, explorers, replay_buffers, utils
+from pfrl.initializers import init_chainer_default
+from pfrl.q_functions import DiscreteActionValueHead
+
+
+def reward_fn(dg, ag):
+    return -1.0 if (ag != dg).any() else 0.0
+
+
+class BitFlip(gym.GoalEnv):
+    """BitFlip environment from https://arxiv.org/pdf/1707.01495.pdf
+
+    Args:
+        n: State space is {0,1}^n
+    """
+
+    def __init__(self, n):
+        self.n = n
+        self.action_space = spaces.Discrete(n)
+        self.observation_space = spaces.Dict(
+            dict(
+                desired_goal=spaces.MultiBinary(n),
+                achieved_goal=spaces.MultiBinary(n),
+                observation=spaces.MultiBinary(n),
+            )
+        )
+        self.clear_statistics()
+
+    def compute_reward(self, achieved_goal, desired_goal, info):
+        return reward_fn(desired_goal, achieved_goal)
+
+    def _check_done(self):
+        success = (
+            self.observation["desired_goal"] == self.observation["achieved_goal"]
+        ).all()
+        return (self.steps >= self.n) or success, success
```

Comment:
I think I should use the wrapper `ContinuingTimeLimit` actually. Exceeding the number of timesteps should not be a terminal signal.